src/index.js

```diff
@@ -62,11 +62,18 @@ export default function testStorySnapshots (options = {}) {
           it(story.name, () => {
             const context = { kind: group.kind, story: story.name }
             const renderedStory = story.render(context)
-            const tree = renderer.create(renderedStory).toJSON()
+            const tree = renderer.create(renderedStory, getTestRendererOptions(story)).toJSON()
             expect(tree).toMatchSnapshot()
           })
         }
       })
     })
   }
 }
+
+const getRenderOptions = story => (story && story.render && story.render.options) ? story.render.options : {}
+
+const getTestRendererOptions = (
+  story,
+  options = getRenderOptions(story)
+) => (options.createNodeMock) ? { createNodeMock: options.createNodeMock } : null
```

Comment:
I don't think hoisting is an issue here as `getTestRenderOptions` is defined before use in the exported `testStorySnapshots` function.

<p data-height="197" data-theme-id="0" data-slug-hash="ZKeayj" data-default-tab="js,result" data-user="theinterned" data-embed-version="2" data-pen-title="hoisted arrow funciton" class="codepen">See the Pen <a href="http://codepen.io/theinterned/pen/ZKeayj/">hoisted arrow funciton</a> by Ned Schwartz (<a href="http://codepen.io/theinterned">@theinterned</a>) on <a href="http://codepen.io">CodePen</a>.</p>
<script async src="https://production-assets.codepen.io/assets/embed/ei.js"></script>